src/Microsoft.AspNet.Server.Kestrel/Http/Frame.cs

```diff
@@ -355,18 +419,8 @@ public void ProduceStart(bool immediate = true, bool appCompleted = false)
             var status = ReasonPhrases.ToStatus(StatusCode, ReasonPhrase);
 
             var responseHeader = CreateResponseHeader(status, appCompleted, ResponseHeaders);
-            SocketOutput.Write(
-                responseHeader.Item1,
-                (error, state) =>
-                {
-                    if (error != null)
-                    {
-                        Log.LogError("ProduceStart ", error);
-                    }
-                    ((IDisposable)state).Dispose();
-                },
-                responseHeader.Item2,
-                immediate: immediate);
+            SocketOutput.Write(responseHeader.Item1, immediate: immediate);
```

Comment:
It'll only block in the case that the response data in write-behind is over 64k. That should be the less likely case, but in that case it's more important to stop allocating unboundedly.